nix/overlay.nix

```
@@ -0,0 +1,36 @@
+# Override some packages and utilities in 'pkgs'
+# and make them available globally via callPackage.
+#
+# For more details see:
+# - https://nixos.wiki/wiki/Overlays
+# - https://nixos.org/nixos/nix-pills/callpackage-design-pattern.html
+prev: final:
```

This is wrong. See doc at https://nixos.wiki/wiki/Overlays :
```suggestion
final: prev:
```
That's the source of your infinite recursion error.